torchgeo/datasets/satlas.py

```diff
@@ -0,0 +1,700 @@
+            'https://ai2-public-datasets.s3.amazonaws.com/satlas/satlas-dataset-v1-labels-static.tar',
+        ),
+        'metadata': (
+            'https://ai2-public-datasets.s3.amazonaws.com/satlas/satlas-dataset-v1-metadata.tar',
+        ),
+    }
+    # TODO
+    md5s: ClassVar[dict[str, tuple[str, ...]]] = {
+        'landsat': (),
+        'naip': (),
+        'sentinel1': (),
+        'sentinel2': (),
+        'labels': (),
+        'metadata': (),
+    }
+
+    # NOTE: 'tci' is RGB (b04-02), not BGR (b02-04)
+    bands: ClassVar[dict[str, tuple[str, ...]]] = {
+        'landsat': tuple(f'b{i}' for i in range(1, 12)),
+        'naip': ('tci', 'ir'),
+        'sentinel1': ('vh', 'vv'),
+        'sentinel2': ('tci', 'b05', 'b06', 'b07', 'b08', 'b11', 'b12'),
+    }
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: str = 'train_lowres',
+        images: Iterable[str] = ('sentinel1', 'sentinel2', 'landsat'),
+        labels: Iterable[str] = ('land_cover',),
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new SatlasPretrain instance.
+
+        Args:
+            root: Root directory where dataset can be found.
+            split: Metadata split to load.
+            images: List of image products.
+            labels: List of label products.
+            transforms: A function/transform that takes input sample and its target as
+                entry and returns a transformed version.
+            download: If True, download dataset and store it in the root directory.
+            checksum: If True, check the MD5 of the downloaded files (may be slow).
+
+        Raises:
+            AssertionError: If *images* is invalid.
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+        """
+        assert set(images) <= set(self.bands.keys())
+
+        self.root = root
+        self.split = split
+        self.images = images
+        self.labels = labels
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+
+        with open(os.path.join(root, 'metadata', f'{split}.json')) as f:
+            self.data = torch.tensor(json.load(f))
+
+    def __len__(self) -> int:
+        """Return the number of images in the dataset.
+
+        Returns:
+            Length of the dataset
+        """
+        return len(self.data)
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: Index to return.
+
+        Returns:
+            Data and label at that index.
+        """
+        col, row = self.data[index]
+        sample: dict[str, Tensor] = {}
+
+        for image in self.images:
+            sample[f'image_{image}'] = self._load_image(image, col, row)
```

Comment:
The design decision here is to use `image_{landsat,naip,sentinel}` as the key so we can retain support for `kornia.augmentation.AugmentationSequential` auto-detecting the type of key/value pairs. Not sure how important this actually is since many augmentations like Normalize will be unique to each image, but it could be useful for augmentations like RandomCrop.